lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,13 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if Regex.match?(~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/, info) do
+      Mix.raise """
+      Your #{lockfile} contains merge conflicts.
+      Run `git checkout HEAD -- #{lockfile}` first to get a clean lock.
```

Comment:
Let's not use `"""` because it adds an uncessary new line at the end. What do you think about this error messages:

```elixir
Mix.raise "Your #{lockfile} contains merge conflicts. If you are using git, " <>
          "run `git checkout HEAD -- #{lockfile}` to revert local changes"
```

?